keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault. There are two ways to store a Secret:
+- setSecret creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+
+secretAsyncClient.setSecret("secret_name", "secret_value").subscribe(secretResponse ->
+  System.out.printf("Secret is created with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value()));
+```
+
+### Retrieve a Secret
+
+Retrieve a previously stored Secret by calling getSecret.
+```Java
+SecretClient secretAsyncClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+        
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse ->
+  System.out.printf("Secret with name %s , value %s \n", secretResponse.value().name(),
+  secretResponse.value().value()));
+```
+
+### Update an existing Secret
+
+Update an existing Secret by calling updateSecret.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse -> {
+     Secret secret = secretResponse.value();
+     //Update the expiry time of the secret.
+     secret.expires(OffsetDateTime.now().plusDays(50));
+     secretAsyncClient.updateSecret(secret).subscribe(secretResponse ->
+         System.out.printf("Secret's updated not before time %s \n", secretResponse.value().notBefore().toString()));
+   });
+```
+
+### Delete a Secret
+
+Delete an existing Secret by calling deleteSecret.
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+        
+secretAsyncClient.deleteSecret("secretName").subscribe(deletedSecretResponse ->
+   System.out.printf("Deleted Secret's deletion time %s \n", deletedSecretResponse.value().deletedDate().toString()));
+```
+
+## Troubleshooting
+
+### General
+
+When you interact with Azure Key Vault Secrets service using this Java client library, errors returned by the service correspond to the same HTTP status codes returned for [REST API][azkeyvault_rest] requests. For example, if you try to retrieve a Secret that doesn't exist in your Key Vault, a `404` error is returned, indicating `Not Found`.
+
+## Contributing
+
+If you would like to become an active contributor to this project please follow the instructions provided in [Microsoft Azure Projects Contribution Guidelines](http://azure.github.io/guidelines.html).
+
+1. Fork it
```

Comment:
Not sure this is actually needed.... I personally like more what Python added

Reply:
Synced with python version